vnext/Microsoft.ReactNative/Fabric/ComponentView.h

```diff
@@ -0,0 +1,48 @@
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#pragma warning(push)
+#pragma warning(disable : 4244)
+#include <react/renderer/componentregistry/ComponentDescriptorProvider.h>
+#include <react/renderer/core/LayoutMetrics.h>
+#pragma warning(pop)
+
+namespace Microsoft::ReactNative {
+
+using RNComponentViewUpdateMask = std::uint_fast8_t;
+/*
+ * Bitmask for all types of possible updates performed during mounting.
+ */
+constexpr RNComponentViewUpdateMask RNComponentViewUpdateMaskNone{0};
+constexpr RNComponentViewUpdateMask RNComponentViewUpdateMaskProps{1 << 0};
+constexpr RNComponentViewUpdateMask RNComponentViewUpdateMaskEventEmitter{1 << 1};
+constexpr RNComponentViewUpdateMask RNComponentViewUpdateMaskState{1 << 2};
+constexpr RNComponentViewUpdateMask RNComponentViewUpdateMaskLayoutMetrics{1 << 3};
+
+constexpr RNComponentViewUpdateMask RNComponentViewUpdateMaskAll = RNComponentViewUpdateMaskProps |
+    RNComponentViewUpdateMaskEventEmitter | RNComponentViewUpdateMaskState | RNComponentViewUpdateMaskLayoutMetrics;
+
+struct IComponentView {
+  virtual std::vector<facebook::react::ComponentDescriptorProvider>
+  supplementalComponentDescriptorProviders() noexcept = 0;
+  virtual void mountChildComponentView(const IComponentView &childComponentView, uint32_t index) noexcept = 0;
```

Comment:
Kind of a general nit, but it would be nice to standardize on PascalCase vs CamelCase for some of these. I know we usually use PascalCase, and FB uses camelCase, but I've definitely seem some funky things around the boundary like a mix of both casing on the same object.